app/controllers/uploads_controller.rb

```diff
@@ -36,7 +36,13 @@ def unzip(dest_folder_name, datafile)
     Dir.mkdir(dest_folder_name)
     Dir.chdir(dest_folder_name) do
       reader.each_entry do |entry|
-        reader.extract(entry, flags.to_i)
+        # For non-directory files, strips the path and extracts to current directory
```

Comment:
Will this flatten *all* directory structure inside the zipfile, or just the top level?

Reply:
This flattens all directory structure.

Reply:
Hm - lots of models have subdirectories that people want to keep. We should just drop the top-level directory if there's just a single one there.

Reply:
Updated code to only change the structure if there is a single directory at the root of the archive. Added four archives to use for testing.